lib/auth/googleauth.js

```diff
@@ -294,12 +309,25 @@ GoogleAuth.prototype.fromJSON = function(json, opt_callback) {
  */
 GoogleAuth.prototype.fromStream = function(stream, opt_callback) {
   var that = this;
-  var jwtClient = new that.JWTClient();
-  jwtClient.fromStream(stream, function(err) {
-    if (err) {
+  if (!stream) {
+    process.nextTick(function() {
+      callback(
+        opt_callback,
+        new Error('Must pass in a stream containing the service account auth settings.'));
+    });
+    return;
+  }
+  var s = '';
```

Comment:
It doesn't make sense to put it in `authclient` or `oauthclient` because `OAuth2Client` does not have anything reasonable to do with a stream, and neither does `computeclient` (another subclass of `Auth2Client`.